packages/typespec-client-generator-core/test/types/array-types.test.ts

```diff
@@ -9,7 +9,16 @@ describe("typespec-client-generator-core: array types", () => {
   beforeEach(async () => {
     runner = await createSdkTestRunner({ emitterName: "@azure-tools/typespec-java" });
   });
-
+  afterEach(async () => {
+    for (const modelsOrEnums of [
+      runner.context.sdkPackage.models,
+      runner.context.sdkPackage.enums,
+    ]) {
+      for (const item of modelsOrEnums) {
```

Comment:
i don't think we should test it this way. Instead, we should add the test on each test and make sure that the generated name is what we expect it to be. We shouldn't just care that its generated name is not empty

Reply:
Agree that devs shall add check for generated name while devs may forget to add the check so this logic is just last Insurance to avoid similar issues.

Reply:
It is here https://github.com/Azure/typespec-azure/pull/1491/files#r1756100153.